core_data_modules/views/traced_data_items_view.py

```diff
@@ -0,0 +1,52 @@
+import collections
+import six
+
+
+# noinspection PyProtectedMember
+class _TracedDataItemsIterator(collections.Iterator):
```

Comment:
I'm a bit queasy about the amount of code duplication here. Is it possible to separate the visitor strategy from the data being extracted?

Reply:
Turns out I actually only need one of these. 

collections can do the rest, so TracedData has been updated to take advantage of that.